listenbrainz/webserver/static/js/src/LastFMImporter.test.tsx

```diff
@@ -368,6 +369,75 @@ describe("submitPage", () => {
   });
 });
 
+describe("getUserPrivacy", () => {
+  beforeEach(() => {
+    const wrapper = shallow<LastFmImporter>(<LastFmImporter {...props} />);
+    instance = wrapper.instance();
+    instance.setState({ lastfmUsername: "dummyUser" });
+
+    // Needed for startImport
+    instance.APIService.getLatestImport = jest.fn().mockImplementation(() => {
+      return Promise.resolve(0);
+    });
+    window.fetch = jest.fn().mockImplementationOnce(() => {
+      return Promise.resolve({
+        ok: true,
+        json: () => Promise.resolve(getInfo),
+      });
+    });
+  });
+
+  it("should call with the correct url", () => {
+    instance.getUserPrivacy();
+
+    expect(window.fetch).toHaveBeenCalledWith(
+      `${props.lastfmApiUrl}?method=user.getrecenttracks&user=${instance.state.lastfmUsername}&api_key=${props.lastfmApiKey}&format=json`
+    );
+  });
+
+  it("should return user privacy status", async () => {
+    // mock function for fetch (no data.error)
+    window.fetch = jest.fn().mockImplementationOnce(() => {
+      return Promise.resolve({
+        ok: true,
+        json: () => Promise.resolve(page),
+      });
+    });
+    expect(await instance.getUserPrivacy()).toEqual(false);
+
+    // mock function for fetch (data.error = 17)
+    window.fetch = jest.fn().mockImplementationOnce(() => {
+      return Promise.resolve({
+        ok: true,
+        json: () => Promise.resolve(getUserPrivacy),
```

Comment:
Small nitpick, but I think the  `getUserPrivacy` file should be called something like  `lastFMPrivateUser`.
As we keep adding more mocks it'll be handy to have a clearer description.